### ISSUE TITLE
WebSocket Error

### ISSUE THREAD

**Issue by marceloboeira** [balderdashy/sails#2168]
WebSocket connection to 'ws://blew.io/socket.io/1/websocket/sFZjWgpMcwONyyml4UAM?__sails_io_sdk_version=0.10.0&__sails_io_sdk_platform=browser&__sails_io_sdk_language=javascript' failed: Error during WebSocket handshake: Unexpected response code: 503 

Does anyone know why this is happening? 

My code: https://github.com/vimia/blew

**Comment by auchenberg** [comment]
Hi,

I'm facing this issue on a Heroku deployment too. Did you find a solution?

**Comment by marceloboeira** [comment]
@auchenberg  nope :/

**Comment by Dallas62** [comment]
Hi,
Do you have uncomment the line `origins` in `config/sockets.js`?
and if it's working, add your domain.

**Comment by auchenberg** [comment]
@Dallas62 No luck. Outcommenting the ``origins`` property still results in:

Client:
`` WebSocket connection to '<domain>/socket.io/1/websocket/IXSxYC_Z-Ahz2OAhf83-?__sails_io_sdk_version=0.10.0&__sails_io_sdk_platform=browser&__sails_io_sdk_language=javascript' failed: Error during WebSocket handshake: Unexpected response code: 503``

Server:
``
2015-01-18T14:43:51.716759+00:00 heroku[router]: at=error code=H13 desc="Connection closed without response" method=GET path="/socket.io/1/websocket/IXSxYC_Z-Ahz2OAhf83-?__sails_io_sdk_version=0.10.0&__sails_io_sdk_platform=browser&__sails_io_sdk_language=javascript" host=<domain> request_id=e5bf66ca-9489-4946-a549-2ea47617ea79 fwd="83.93.30.73,141.101.80.16" dyno=web.1 connect=0ms service=4ms status=503 bytes=0
``

**Comment by Dallas62** [comment]
And of you modify `config/cors.js` to allow it?

If that doesn't work, this could be the heroku router that block you when you come from another domain.

**Comment by auchenberg** [comment]
CORS is enabled for all routes, so this must be somewhat related to Heroku's routing.

**Comment by Dallas62** [comment]
This one speak about reactivating:
> http://stackoverflow.com/questions/19668218/websocket-not-working-with-socket-io-in-heroku/19756090#19756090

`heroku config:set DOMAIN=customDomainChoosed.herokuapp.com`
> https://github.com/DaftMonk/generator-angular-fullstack/issues/395

But I'm using AWS and HAProxy, as revert proxy ,for mine projects, so... can't do more...

**Comment by auchenberg** [comment]
Yeah, the heroku labs feature is deprecated, as websockets now are available to everyone, so that's not an option. I suspect this to be a socket.io + heroku problem, so there's probably others who have this problem.

**Comment by uncletammy** [comment]
I'm also having this problem with socket.io on a Heroku deployment . Has anyone solved this?  Is it just me or do these errors seems to correlate with specific ISP routing nodes?

**Comment by stenio123** [comment]
hey guys I was having the same problem, and it seems just adding the following lines should work
     proxy_http_version 1.1;
     proxy_set_header Upgrade $http_upgrade;
     proxy_set_header Connection "upgrade";

However please remember to also do the following:
1) are you placing the configuration inside etc/nginx/sites-available? Make sure that the one on etc/nginx/sites-enabled matches exactly.
2) did you reload nginx after the changes? run "nginx -s reload" on the server

This resolved the issue for me, hope it helps!

**Comment by fettay** [comment]
Hi guys,

**Comment by fettay** [comment]
Hi guys, I just faced the same problem , the solution was to uncomment the line :"  transports: ["polling", "websocket"]," on config/sockets.js. Good luck !

**Comment by sailsbot** [comment]
Thanks for posting, @marceloboeira.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
